tutor/commands/plugins.py

```diff
@@ -105,48 +171,279 @@ def disable(context: Context, plugin_names: list[str]) -> None:
+            plugin = indexes.find_in_cache(name)
+            src = hooks.Filters.PLUGIN_INDEX_ENTRY_TO_INSTALL.apply(plugin.data)[
+                "src"
+            ].strip()
+            if utils.is_url(src):
+                single_file_plugins.append(src)
+            else:
+                # Create requirements file where each plugin reqs is prefixed by a
+                # comment with its name
+                pip_requirements.append(f"# {name}\n{src}")
+
+    for url in single_file_plugins:
+        install_single_file_plugin(url)
+
+    if pip_requirements:
+        # pip install -r reqs.txt
+        requirements_txt = "\n".join(pip_requirements)
+        with tempfile.NamedTemporaryFile(
+            prefix="tutor-reqs-", suffix=".txt", mode="w"
+        ) as tmp_reqs:
+            tmp_reqs.write(requirements_txt)
+            tmp_reqs.flush()
+            fmt.echo_info(f"Installing pip requirements:\n{requirements_txt}")
+            utils.execute(
+                "pip", "install", *pip_install_opts, "--requirement", tmp_reqs.name
+            )
+
+
+def install_single_file_plugin(location: str) -> None:
+    """
+    Download or copy a single file to the plugins root.
+    """
+    plugin_path = os.path.join(PLUGINS_ROOT, os.path.basename(location))
+    if not plugin_path.endswith(".yml") and not plugin_path.endswith(".py"):
+        plugin_path += ".py"
+    # Read url
+    fmt.echo_info(f"Downloading plugin from {location}...")
+    content = utils.read_url(location)
     # Save file
-    if not os.path.exists(PLUGINS_ROOT):
-        os.makedirs(PLUGINS_ROOT)
+    utils.ensure_file_directory_exists(plugin_path)
     with open(plugin_path, "w", newline="\n", encoding="utf-8") as f:
         f.write(content)
     fmt.echo_info(f"Plugin installed at {plugin_path}")
 
 
+@click.command()
+@click.argument("pattern", default="")
+def search(pattern: str) -> None:
+    """
+    Search in plugin descriptions.
+    """
+    results: list[tuple[str, ...]] = [("NAME", "STATUS", "DESCRIPTION")]
+    for plugin in indexes.iter_cache_entries():
+        if plugin.match(pattern):
+            results.append(
+                (
+                    plugin.name,
+                    plugin_status(plugin.name),
+                    plugin.short_description,
+                )
+            )
+    print(format_table(results))
+
+
+@click.command()
+@click.argument("name", type=IndexPluginName())
+def show(name: str) -> None:
+    """
+    Show plugin details from index.
+    """
+    name = name.lower()
+    for plugin in indexes.iter_cache_entries():
+        if plugin.name == name:
+            fmt.echo(
+                f"""Name: {plugin.name}
+Source: {plugin.src}
+Status: {plugin_status(name)}
+Author: {plugin.author}
+Maintainer: {plugin.maintainer}
+Homepage: {plugin.url}
+Index: {plugin.index}
+Description: {plugin.description}"""
+            )
+            return
+    raise exceptions.TutorError(f"Plugin could not be found: {name}")
```

Comment:
```suggestion
    raise exceptions.TutorError(f"Plugin could not be found in indexes: {name}")
```

If I run this on a locally-installed plugin that's not in an index, then I will see this error. I think it'd be clearer to be told that the plugin is "not found in an index" than "not found".